instant-segment/src/lib.rs

```diff
@@ -198,6 +200,59 @@ impl<'a> SegmentState<'a> {
         }
 
         self.search.result.reverse();
+        score
+    }
+}
+
+/// The result of segmenting a sentence.
+pub struct Segments<'a> {
+    segments: &'a [String],
+    score: f64,
+}
+
+impl<'a> Segments<'a> {
+    fn new(segments: &'a [String], score: f64) -> Self {
+        Self { segments, score }
+    }
+
+    /// Returns the score of the segmented sentence.
+    pub fn score(&self) -> f64 {
+        self.score
+    }
+
+    /// Returns an iterator over the segmented words of a sentence.
+    pub fn iter(&self) -> SegmentsIter<'a> {
+        SegmentsIter {
+            segments_iter: self.segments.iter(),
+        }
+    }
```

Comment:
These seem like orthogonal questions?  We could derive Clone and Copy either way, but is it useful for this type?

`IntoIterator` is quite nice because it allows writing `for word in &results { ... }` 

Perhaps I have misunderstood the suggestion.  It's your project, feel free to push an arrangement that you like better!  The main thing I am asking for is the ability to get the score directly from the original segmentation, without having to re-calculate it.